mcproto/types/nbt.py

```diff
@@ -0,0 +1,1399 @@
+            will be used.
+        :param with_name: Whether to read the name of the tag to the buffer as a TAG_String.
+
+        :return: The LongNBT tag.
+        """
+        name, tag_type = cls._read_header(buf, read_type=with_type, with_name=with_name)
+        if _get_tag_type(cls) != tag_type:
+            raise TypeError(f"Expected a {_get_tag_type(cls).name} tag, but found a different tag ({tag_type.name}).")
+
+        if buf.remaining < 8:
+            raise IOError("Buffer does not contain enough data to read a long.")
+
+        payload = int.from_bytes(buf.read(8), "big", signed=True)
+        return LongNBT(payload, name=name)
+
+
+class FloatNBT(NBTag):
+    """NBT tag representing a floating-point value, represented as a 32-bit IEEE 754-2008 binary32 value."""
+
+    payload: float
+
+    __slots__ = ()
+
+    @override
+    def write_to(self, buf: Buffer, with_type: bool = True, with_name: bool = True) -> None:
+        """Write the FloatNBT tag to the buffer.
+
+        :param buf: The buffer to write to.
+        :param with_type: Whether to include the type of the tag in the serialization.
+        :param with_name: Whether to include the name of the tag in the serialization.
+
+        :note: The float value is written as a 32-bit floating-point value in big-endian format.
+        """
+        self._write_header(buf, with_type=with_type, with_name=with_name)
+        buf.write_value(StructFormat.FLOAT, self.payload)
+
+    @override
+    @classmethod
+    def read_from(cls, buf: Buffer, with_type: bool = True, with_name: bool = True) -> FloatNBT:
+        """Read the FloatNBT tag from the buffer.
+
+        :param buf: The buffer to read from.
+        :param with_type: Whether to read the type of the tag from the buffer. If this is False, the type of the class
+            will be used.
+        :param with_name: Whether to read the name of the tag to the buffer as a TAG_String.
+
+        :return: The FloatNBT tag.
+        """
+        name, tag_type = cls._read_header(buf, read_type=with_type, with_name=with_name)
+        if _get_tag_type(cls) != tag_type:
+            raise TypeError(f"Expected a {_get_tag_type(cls).name} tag, but found a different tag ({tag_type.name}).")
+
+        if buf.remaining < 4:
+            raise IOError("Buffer does not contain enough data to read a float.")
+
+        return FloatNBT(buf.read_value(StructFormat.FLOAT), name=name)
+
+    def __float__(self) -> float:
+        """Get the float value of the FloatNBT tag."""
+        return self.payload
+
+    @override
+    def __eq__(self, other: object) -> bool:
+        """Check equality between two FloatNBT tags.
+
+        :param other: The other FloatNBT tag to compare to.
+
+        :return: True if the tags are equal, False otherwise.
+
+        :note: The float values are compared with a small epsilon (1e-6) to account for floating-point errors.
```

Comment:
Should we really be accounting for this? Why is that our task? `float` type doesn't do this in it's `__eq__`.